File: dtos/ingredient_dto.go
package dtos

type IngredientDto struct {
	ID         uint      `json:"id"`
	Name       string    `json:"name"`
	ParentID   *uint     `json:"parentId"`
	IsAllergen bool      `json:"isAllergen"`
	Type       TypeDto   `json:"type"`
	Image      *ImageDto `json:"image"`
}

File: web/src/data/dtos/ingredients/IngredientDto.ts
import { IngredientType } from '@/shared/enums/IngredientType'
import { ImageDto } from './ImageDto'
import type { IType } from './IType'

export class IngredientDto {
  id: number = 0
  name: string = ''
  parentId: number | null = null

  type: IType<IngredientType> = {
    name: '',
    code: IngredientType.OTHER,
  }

  isAllergen: boolean = false

  image: ImageDto = new ImageDto({})

  constructor(obj: Partial<IngredientDto>) {
    Object.assign(this, obj)
    this.image = obj.image ? new ImageDto(obj.image) : new ImageDto({})
  }
}
